config/core/build-configs-android.yaml

```
@@ -41,6 +41,25 @@ android_variants: &android_variants
             - 'allmodconfig'
             - 'allnoconfig'
 
+  clang-14:
+    build_environment: clang-14
+    architectures:
+        arm: *arm_arch
+        arm64:
+          extra_configs:
+            - 'allmodconfig'
+            - 'allnoconfig'
+            - 'defconfig+CONFIG_CPU_BIG_ENDIAN=y'
+            - 'defconfig+CONFIG_RANDOMIZE_BASE=y'
+            - 'gki_config'
+        i386: *i386_arch
+        riscv: *riscv_arch
+        x86_64:
+          base_defconfig: 'x86_64_defconfig'
+          extra_configs:
+            - 'allmodconfig'
+            - 'allnoconfig'
+            - 'gki_config'
```

It's a bit better to inherit from the definition in `gcc-10` and override fields that change, for example:
```suggestion
        x86_64:
          <<: * x86_64_arch
          extra_configs:
            - 'allmodconfig'
            - 'allnoconfig'
            - 'gki_config'
```
The same can be done for `arm64` even though there's nothing to inherit.  It makes the config easier to maintain as future changes in the `gcc-10` section that are architecture-specific would be automatically propagated to the other sections such as `clang-14`.